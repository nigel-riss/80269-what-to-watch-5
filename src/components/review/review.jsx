import React, {PureComponent} from 'react';
import {Link} from 'react-router-dom';
import filmType from '../../types/film.js';
import Logo from '../logo/logo.jsx';


const DEFAULT_RATING = 3;

class Review extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      rating: DEFAULT_RATING,
      text: ``,
    };

    this._handleTextareaChange = this._handleTextareaChange.bind(this);
    this._handleRatingChange = this._handleRatingChange.bind(this);
  }

  _handleTextareaChange(e) {
    this.setState({
      text: e.target.value,
    });
  }

  _handleRatingChange(rating) {
    this.setState({
      rating,
    });
  }

  render() {
    const {
      cover,
      poster,
      name,
    } = this.props.film;

    const {
      rating,
      text,
    } = this.state;

    return (
      <section className="movie-card movie-card--full">
        <div className="movie-card__header">
          <div className="movie-card__bg">
            <img src={cover} alt={name} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <header className="page-header">
            <Logo/>

            <nav className="breadcrumbs">
              <ul className="breadcrumbs__list">
                <li className="breadcrumbs__item">
                  <Link
                    to="/films/1"
                    className="breadcrumbs__link"
                  >
                    {name}
                  </Link>
                </li>
                <li className="breadcrumbs__item">
                  <a className="breadcrumbs__link">Add review</a>
                </li>
              </ul>
            </nav>

            <div className="user-block">
              <div className="user-block__avatar">
                <img src="/img/avatar.jpg" alt="User avatar" width="63" height="63" />
              </div>
            </div>
          </header>

          <div className="movie-card__poster movie-card__poster--small">
            <img
              src={poster}
              alt={`${name} poster`}
              width="218"
              height="327"
            />
          </div>
        </div>

        <div className="add-review">
          <form
            action="#"
            className="add-review__form"
            onSubmit={(e) => {
              e.preventDefault();
            }}
          >
            <div className="rating">
              <div className="rating__stars">
                {Array(5).fill({}).map((it, i) => <React.Fragment key={i}>
                  <input
                    className="rating__input"
                    id={`star-${i + 1}`}
                    type="radio"
                    name="rating"
                    value={i + 1}
                    checked={rating === i + 1}
                    onChange={() => {
                      this._handleRatingChange(i + 1);
                    }}
                  />
                  <label
                    className="rating__label"
                    htmlFor={`star-${i + 1}`}
                  >
                    Rating {i + 1}
                  </label>
                </React.Fragment>)}
              </div>
            </div>

            <div className="add-review__text">
              <textarea
                className="add-review__textarea"
                name="review-text"
                id="review-text"
                placeholder="Review text"
                value={text}
                onChange={this._handleTextareaChange}
              ></textarea>
              <div className="add-review__submit">
                <button className="add-review__btn" type="submit">Post</button>
              </div>
            </div>
          </form>
        </div>

      </section>
    );
  }
}


Review.propTypes = filmType;


export default Review;
