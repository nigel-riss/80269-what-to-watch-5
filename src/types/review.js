import PropTypes from 'prop-types';


export default PropTypes.shape({
  date: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  text: PropTypes.number.isRequired,
});
