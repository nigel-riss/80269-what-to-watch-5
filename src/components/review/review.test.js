import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import {Review} from './review.jsx';
import filmsMock from '../../mocks/films.js';
import storeMock from '../../mocks/store';


const mockStore = configureMockStore([]);
const store = mockStore(storeMock);
const noop = () => {};

it(`Should Review render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <Review
              currentFilm={filmsMock[0]}
              isReviewInputLocked={false}
              getFilm={noop}
              filmId={1}
              onFormSubmit={noop}
              renderForm={noop}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
