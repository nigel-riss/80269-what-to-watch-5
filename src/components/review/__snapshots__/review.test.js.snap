// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Review render correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Aviator"
        src="/img/aviator.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              The Aviator
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Aviator poster"
        height="327"
        src="/img/aviator-poster.png"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  />
</section>
`;
